internal/index/scip.go

```diff
@@ -131,6 +131,9 @@ func Index(writer func(proto.Message), opts config.IndexOpts) error {
 					continue
 				}
 
+				enclosingNodeMap := map[*ast.Ident]ast.Node{}
+				visitors.EnclosingNodeMap(f, enclosingNodeMap)
```

Comment:
Two "maybe"s:
1) Maybe it's a good idea to move `enclosingNodeMap` initialization to `EnclosingNodeMap`, so that we can have `enclosingNodeMap := visitors.EnclosingNodeMap(...)` (or inline this inside `NewFileVisitor`)
2) Maybe it makes sense to actually move the `EnclosingNodeMap` call inside `visitors.NewFileVisitor` and make the function private? I agree that having O(n) logic inside structure initialization is usually a bad habit but on the other hand this is the only use case.